RELEASE/scripts/autoscend/quests/level_11.ash

```diff
@@ -1,5 +1,105 @@
 script "level_11.ash"
 
+
+record desert_buff_record
+{
+	item offhand;
+	item fam_equip;
+	familiar fam;
+	int progress;
+};
+
+desert_buff_record desertBuffs()
+{
+    desert_buff_record dbr;
+
+    dbr.progress = 1;
+
+	boolean possessUnrestricted(item it)
+	{
+		return possessEquipment(it) && is_unrestricted(it);
+	}
+
+	boolean compassValid = possessUnrestricted($item[UV-resistant compass]);
+	boolean lhmValid = canChangeToFamiliar($familiar[Left-Hand Man]);
+	boolean meloValid = canChangeToFamiliar($familiar[Melodramedary]);
+	boolean odrValid = possessUnrestricted($item[Ornate Dowsing Rod]);
+
+	dbr.fam = $familiar[none];
+	dbr.fam_equip = $item[none];
+	dbr.offhand = $item[none];
+
+	// If we can't use the Ornate dowsing rod
+	if (!odrValid)
+	{
+		// And we can use the compass
+		if (compassValid)
+		{
+			// And we have the Left-Hand man but not the Melodramedary
+			// Free up our offhand for something useful
+			if (lhmValid && !meloValid)
+			{
+				dbr.fam = $familiar[Left-Hand Man];
+				dbr.fam_equip = $item[UV-resistant compass];
+				dbr.progress += 1;
+			}
+			// Otherwise hold the compass
+			else
+			{
+				dbr.offhand = $item[UV-resistant compass];
+				dbr.progress += 1;
+			}
+		}
+
+		// If we have the Melodramedary use it!
+		if (meloValid)
+		{
+			dbr.fam = $familiar[Melodramedary];
+			dbr.progress += 1;
+		}
+	}
+	// Otherwise
+	else
+	{
+		// If we have it and a Left-Hand man is our best familiar choice
+		// but we have no compass free up our offhand
+		if (!compassValid && lhmValid && !meloValid)
+		{
+			dbr.fam = $familiar[Left-Hand Man];
+			dbr.fam_equip = $item[Ornate Dowsing Rod];
+			dbr.progress += 2;
+		}
+		// Otherwise we can just hold it
+		else
+		{
+			dbr.offhand = $item[Ornate Dowsing Rod];
+			dbr.progress += 2;
+		}
+
+		// Melodramedary is better here though
+		if (meloValid)
+		{
+			dbr.fam = $familiar[Melodramedary];
+			dbr.progress += 1;
+		}
+		// Otherwise we can give the compass to the Left-Hand man if possible
+		else if (compassValid && lhmValid)
+		{
+			dbr.fam = $familiar[Left-Hand Man];
+			dbr.fam_equip = $item[UV-resistant compass];
+			dbr.progress += 1;
+		}
+	}
+
+	// There are some other familiars we might choose if nothing affects progress
+	if (dbr.fam == $familiar[none] && canChangeToFamiliar($familiar[Artistic Goth Kid]))
```

Comment:
need to also check if there are free fights remaining for artistic goth kid before using it.
also mini hipster. anyone knows how their shared counter works in case you have both?